include/silo/storage/dictionary.h

```diff
@@ -3,35 +3,38 @@
 
 #include <filesystem>
 #include <iostream>
+#include <optional>
 #include <string>
 #include <unordered_map>
 #include <vector>
 
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+
 namespace silo {
 class PangoLineageAliasLookup;
 
+using ColumnName = std::string;
+using ValueId = uint32_t;
+using PangoLineage = std::string;
+
 class Dictionary {
   private:
-   // TODO(someone) put this into two way map
-   std::unordered_map<std::string, uint32_t> pango_lineage_dictionary;
-   std::unordered_map<std::string, uint32_t> country_dictionary;
-   std::unordered_map<std::string, uint32_t> region_dictionary;
-   std::unordered_map<std::string, uint32_t> additional_columns_dictionary;
-   std::unordered_map<std::string, uint64_t> general_dictionary;  // TODO(someone) clean this up!
-
-   std::vector<std::string> pango_lineage_lookup;
-   std::vector<std::string> country_lookup;
-   std::vector<std::string> region_lookup;
-   std::vector<std::string> additional_columns_lookup;
-   std::vector<std::string> general_lookup;
-
-   uint32_t pango_lineage_count = 0;
-   uint32_t country_count = 0;
-   uint32_t region_count = 0;
-   uint32_t additional_columns_count = 0;
-   uint64_t general_count = 0;
+   std::unordered_map<ColumnName, std::unordered_map<std::string, ValueId>> string_lookup;
+   std::unordered_map<ColumnName, std::vector<std::string>> reverse_string_lookup;
+   void updateStringLookup(const ColumnName& column_name, const std::string& value);
+
+   std::unordered_map<ColumnName, std::unordered_map<PangoLineage, ValueId>> pango_lineage_lookup;
+   std::unordered_map<ColumnName, std::vector<PangoLineage>> reverse_pango_lineage_lookup;
+   void updatePangoLineageLookup(const ColumnName& column_name, const PangoLineage& pango_lineage);
```

Comment:
We could probably also encapsulate this and deduplicate it? Does that make sense? And maybe find a better name than "TypedLookup".
```suggestion
   TypedLookup<std::string> string_lookup;
   TypedLookup<PangoLineage> pango_lineage_lookup;
```

Reply:
Good idea.